src/passport-saml-types.ts

```diff
@@ -0,0 +1,10 @@
+import * as passport from "passport";
```

Comment:
I'm not going to push too hard on such an early PR, but I really don't think this project should have anything `passport`-related in it. What is this for?

Reply:
turned out that there was still one cross dependency in the type definitions. this is handling that for now

Reply:
Sure thing... that is why we should probably release a pair of 4.0. I'm not against this being a rather quick release if we can separate out everything. We had good success with keeping the 2.x branch alive while we sorted out the 3.0 stuff. We can do the same with keeping the 3.x branch alive while we build toward a 4.0 release.

Reply:
agreed. and someone who is deeper into the passport world and typescript should probably look at these type definitions.

Reply:
If we are using `passport` here then why is `passport` removed  from package dependencies.

Reply:
it should be removed from dependencies. it's only here in the types because those types are still somewhat mingled. if someone who's into typescript could clean them up that'd be great. I'm more a JS guy

Reply:
Let's get this landed and then we'll have a clearer picture of what needs to be done to clean this up.